features/netsocket/NetworkInterface.cpp

```diff
@@ -65,3 +65,17 @@ nsapi_error_t NetworkInterface::add_dns_server(const SocketAddress &address)
     return get_stack()->add_dns_server(address);
 }
 
+void NetworkInterface::attach(mbed::Callback<void(nsapi_connection_status_t, int)> status_cb)
+{
+}
+
+nsapi_connection_status_t NetworkInterface::get_connection_status()
+{
+    return NSAPI_STATUS_UNDEFINED;
+}
+
+nsapi_error_t NetworkInterface::set_blocking(bool blocking)
+{
+    return NSAPI_ERROR_UNSUPPORTED;
```

Comment:
If you're adding this, you should implement it for lwIP and Nanostack, which I don't think you are.